compiler/sem/semstmts.nim

```diff
@@ -1967,10 +1967,11 @@ proc typeSectionRightSidePass(c: PContext, n: PNode) =
       # ``static type``
       rawAddSon(s.typ, newTypeS(tyNone, c))
       s.ast = a
-      inc c.inGenericContext
-      var body = semTypeNode(c, a[2], nil)
-      dec c.inGenericContext
-      if body != nil:
+      # magic type definitions are allowed to not have a body on the right
+      if s.magic == mNone or a[2].kind != nkEmpty:
```

Comment:
This is a reasonable approach, at best we could make `semTypeNode` treat this as a `tyNone`, or something, but that could a) lead to other issues given the current sloppiness in the compiler, and b) might not be all that principled and still require special casing here.